src/coingecko/Coingecko.e2e.ts

```diff
@@ -1,17 +1,41 @@
 import assert from "assert";
-import Coingecko from "./Coingecko";
 import dotenv from "dotenv";
 import winston from "winston";
+import * as coingecko from "./Coingecko";
 dotenv.config({ path: ".env" });
 
+const Coingecko = coingecko.Coingecko;
+const msToS = coingecko.msToS;
+type CoinGeckoPrice = coingecko.CoinGeckoPrice;
+
 const dummyLogger = winston.createLogger({
   level: "debug",
   transports: [new winston.transports.Console()],
 });
 
+class TestGecko extends Coingecko {
+
+  private static testInstance: TestGecko | undefined;
+
+  public static get(logger: winston.Logger) {
+    if (!this.testInstance)
+      this.testInstance = new TestGecko(logger);
+    return this.testInstance;
+  }
+
+  // Hack to return protected getPriceCache.
+  _getPriceCache(currency: string, platform_id: string): { [addr: string]: CoinGeckoPrice } {
+    return this.getPriceCache(currency, platform_id);
+  }
+
+  constructor(logger: winston.Logger) {
+    super("127.0.0.1", "127.0.0.1", logger);
+  };
+}
+
 // this requires e2e testing, should only test manually for now
 describe("coingecko", function () {
-  let cg: Coingecko;
+  let cg: coingecko.Coingecko;
```

Comment:
@daywiss This one is frustrating me..any suggestions on how to import the Coingecko class in a way that permits it to be used as both a type and class?

Reply:
Yeah, works fine. Was pretty sure that I was previously unable to import CoinGeckoPrice this way, but it seems to work now, so not exactly sure what my issue was. Thanks.